pandas/core/internals/blocks.py

```diff
@@ -1969,6 +1969,9 @@ def convert(
         attempt to cast any object types to better types return a copy of
         the block (if copy = True) by definition we ARE an ObjectBlock!!!!!
         """
+        if self.dtype != object:
+            return [self]
```

Comment:
i think this may need to make a copy in some cases?

Reply:
Are you sure? 
```
ser = Series(["a"], dtype="object")
result = ser.infer_objects()
result.iloc[0] = "c"
print(ser)
```

this updates ser as well, same as the bytes version. If we add a copy, the bytes version would not update anymore

Reply:
1) infer_objects in principle makes a copy in cases where it doesn't do inference. 2) I don't think that's what most users (or our internal usages) actually want.  3) ATM the ObjectBlock case that doesn't do inference doesn't make a copy, which is inconsistent with other dtypes.  4) i think adding a copy keyword makes sense xref #48141, but adding copy keywords in general got pushback recently

Reply:
Can address in follow up